include/revng/TupleTree/TupleTreeDiff.h

```diff
@@ -38,15 +44,58 @@ void addToContainer(C &Container, const typename C::value_type &Value) {
   Container.push_back(Value);
 }
 
+template<typename T>
+struct TupleTreeEntries {
+  using Types = std::variant<char>;
+};
+
+template<typename T>
+using TupleTreeEntriesT = typename TupleTreeEntries<T>::Types;
+
+template<>
+struct TupleTreeEntries<model::Binary> {
+  using Types = model::AllTypes;
+};
```

Comment:
You can't have this here, this file has to be model-agnostic.

Reply:
ah, yeah, forgot to move it